contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -3416,9 +3416,39 @@ LANGUAGE C STABLE PARALLEL SAFE;
 CREATE OR REPLACE FUNCTION sys.is_member(IN role sys.SYSNAME)
 RETURNS INT AS
 $$
-	SELECT sys.is_rolemember_internal(role, NULL);
+BEGIN
+    -- Always return 1 for 'public'
+    IF (role = 'public')
+    THEN RETURN 1;
+    END IF;
+
+    IF EXISTS (SELECT orig_loginname FROM sys.babelfish_authid_login_ext WHERE orig_loginname = role)
+    THEN
+        IF EXISTS (SELECT name FROM sys.login_token WHERE name = role)
+        THEN RETURN 1; -- Return 1 if current session user is a member of role or windows group
```

Comment:
Acc. to TSQL documentation, is_member should not be showing membership of server roles. It will show only database roles and windows group membership(Please double-check).